andyholmes/flatter
# Implement AppStream data support

This implements support for AppStream data provided via [AppData files](https://docs.flatpak.org/en/latest/conventions.html#appdata-files) which is responsible mainly for the information presented in GUI Flatpak managers, but is also used to show the current application version in the CLI as well as making the application findable when running `flatpak search`.

How to test:

1. Build and deploy the repository
2. Run `flatpak update --appstream <repository name>`; this should not produce an error
3. Try searching for an application in your repository that exposes an AppData file via `flatpak search` and see if it's found and if it displays it's current version number

Example how it should look like when it's working:

![image](https://github.com/andyholmes/flatter/assets/80664890/163b5af2-7555-457a-9eeb-d7973026e19c)

The Flatpak manifest for the application shown in this example can be found [here](https://github.com/mserajnik/flatpak.mser.at/blob/master/apps/com.gitlab.martintornqvist.InfraArcana/com.gitlab.martintornqvist.InfraArcana.yaml) and the AppData file [here](https://github.com/mserajnik/flatpak.mser.at/blob/master/apps/com.gitlab.martintornqvist.InfraArcana/com.gitlab.martintornqvist.InfraArcana.metainfo.xml).

Closes #71
